### PR TITLE
Update Alertmanager fork

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR updates the Alertmanager fork version to include the following PRs:
- https://github.com/grafana/prometheus-alertmanager/pull/90
- https://github.com/grafana/prometheus-alertmanager/pull/91
- https://github.com/grafana/prometheus-alertmanager/pull/92

It also updates the Go version we use in CI and the `golangci-lint` version, and fixes linting issues.

Related issue: https://github.com/grafana/grafana/issues/92707

### CODE REVIEW

**Review by titolins — APPROVED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 output:
-  format: line-number
+  formats: line-number
```
Comment (santihernandezc) [inline .golangci.yml:2]: Changes to this file are related to removing deprecated config options in favor of the suggested ones.

File: Makefile [hunk 2]
```diff
@@ -46,4 +46,4 @@ drone: .drone/drone.yml
 	GOPATH=$(CURDIR)/.tools go install github.com/fatih/faillint@v1.10.0
 
 .tools/bin/golangci-lint: .tools
-	GOPATH=$(CURDIR)/.tools go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.55.2
+	GOPATH=$(CURDIR)/.tools go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.61.0
```
Comment (santihernandezc) [inline Makefile:49]: Latest golangci-lint version ([release page](https://github.com/golangci/golangci-lint/releases)).

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,8 @@
 module github.com/grafana/alerting
 
-go 1.21
+go 1.22
```
Comment (santihernandezc) [inline go.mod:3]: Required by the new version of the AM fork.

File: notify/grafana_alertmanager_metrics.go [hunk 4]
```diff
@@ -20,10 +21,10 @@ type GrafanaAlertmanagerMetrics struct {
 }
 
 // NewGrafanaAlertmanagerMetrics creates a set of metrics for the Alertmanager.
-func NewGrafanaAlertmanagerMetrics(r prometheus.Registerer) *GrafanaAlertmanagerMetrics {
+func NewGrafanaAlertmanagerMetrics(r prometheus.Registerer, l log.Logger) *GrafanaAlertmanagerMetrics {
```
Comment (santihernandezc) [inline notify/grafana_alertmanager_metrics.go:24]: The `NewAlerts()` function now takes a logger to log failed collector registration attempts (check [here](https://github.com/grafana/prometheus-alertmanager/blob/b5e64e81e8d3630666a75582716a912a41093ba5/api/metrics/metrics.go#L47)).

File: .golangci.yml [hunk 5]
```diff
@@ -13,7 +13,9 @@ linters-settings:
   errcheck:
     # path to a file containing a list of functions to exclude from checking
     # see https://github.com/kisielk/errcheck#excluding-functions for details
-    exclude: ./.errcheck-exclude
```
Comment (titolins) [inline .golangci.yml:16]: would be nice to remove/update the comment (`path to a file containing a list of functions to exclude from checking`)

**Review by titolins — APPROVED** [review]
Just curious - why updating golanci-lint here? Is it required because of the go update?

File: .golangci.yml (hunk 5 above)
Comment (santihernandezc) [reply .golangci.yml:16]: Good catch! 😅

**Comment by santihernandezc** [comment]
@titolins Yep, updating the Go version required an update in `golangci-lint` too.

**Pull request merged**